[stable/sonarqube] Add MySQL database support to Sonarqube

irajdeep opened the pull request:
<!--
Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

**What this PR does / why we need it**:
This PR adds `MySQL` database support to Sonarqube. Currently Sonarqube has only`Postgresql` support to store the data. It has the relevant changes to the template files and README.md for the new MySQL database support

**Which issue this PR fixes** *
fixes #5654 

**Special notes for your reviewer**:
`/assign`  @rjkernick

irajdeep committed:
Added MySQL database support to Sonarqube, the Sonarqube helm chart now has an option to store data in MySQL database too, It already had a support to store the data in Postgres. A given release could now have either Postgres database (internal / external) or MySQL database (internal / external).

irajdeep committed:
Updated README.md 

Updated README.md to add configuration changes that come with MySQL database support.

irajdeep committed:
Removing white spaces to get rid of helm lint errors

irajdeep committed:
Yes another attempt to remove trailing spaces :(

rjkernick commented on stable/sonarqube/Chart.yaml:
```diff
@@ -1,6 +1,6 @@
 name: sonarqube
 description: Sonarqube is an open sourced code quality scanning tool
-version: 0.6.0
+version: 0.6.1
```
Please bump the minor version since this adds a new feature

irajdeep replied:
Fixed

rjkernick commented on stable/sonarqube/templates/secret.yaml:
```diff
@@ -12,3 +12,18 @@ type: Opaque
+    heritage: {{ .Release.Service }}
+type: Opaque
+data:
+  postgres-password: {{ .Values.mysql.mysqlPassword | b64enc | quote }}
```
Key should be mysql-password

irajdeep replied:
Ah my bad , this is a blunder .

irajdeep replied:
Fixed

rjkernick commented on stable/sonarqube/templates/secret.yaml:
```diff
@@ -12,3 +12,18 @@ type: Opaque
 data:
   postgres-password: {{ .Values.postgresql.postgresPassword | b64enc | quote }}
 {{- end -}}
+{{- if eq .Values.mysql.enabled false -}}
```
We also only need this if the database type is mysql. Same with the postgres secret above only is needed is database is postgres and postgres is disabled.

irajdeep replied:
Added the check , thanks for pointing out

irajdeep committed:
Fixing PR comments
1. Bumped the minor version of the chart since it's a feature release
2. Added check for database type while populating secrets
3. Corrected mysql key

rjkernick commented:
/ok-to-test

rjkernick commented:
I am going to try and test out the different combinations this afternoon before I merge it. Thanks for the quick turnaround on the changes!

irajdeep commented:
Sure, let me know if you find any issues . Thanks !!

rjkernick commented:
/assign

irajdeep commented:
@rjkernick Can we go ahead with merging .

rjkernick commented:
/lgtm
